src/node/gov.h

```diff
@@ -5,26 +5,58 @@
 #include "ds/json.h"
 #include "entities.h"
 #include "kv/map.h"
+#include "proposals.h"
 
 namespace ccf
 {
   namespace jsgov
   {
     using ProposalId = std::string;
-    using Proposal = std::string;
+
     struct ProposalInfo
     {
       ccf::MemberId proposer_id;
+      ccf::ProposalState state;
       std::unordered_map<ccf::MemberId, std::string> ballots = {};
     };
-    DECLARE_JSON_TYPE(ProposalInfo)
-    DECLARE_JSON_REQUIRED_FIELDS(ProposalInfo, proposer_id, ballots);
+    DECLARE_JSON_TYPE(ProposalInfo);
+    DECLARE_JSON_REQUIRED_FIELDS(ProposalInfo, proposer_id, state, ballots);
+
+    struct ProposalSubmitted
+    {
+      ProposalId proposal_id;
+      ccf::ProposalState state;
+    };
+    DECLARE_JSON_TYPE(ProposalSubmitted);
+    DECLARE_JSON_REQUIRED_FIELDS(ProposalSubmitted, proposal_id, state);
 
-    using ProposalMap = kv::RawCopySerialisedMap<ProposalId, Proposal>;
+    using ProposalMap = kv::RawCopySerialisedMap<ProposalId, std::string>;
     using ProposalInfoMap = kv::MapSerialisedWith<
```

Comment:
Use `ccf::ServiceMap`?

Reply:
How do you mean? The types are structurally identical, but that's a coincidence, not a deliberate choice.

Reply:
Gotcha